src/main/java/org/opentripplanner/routing/core/ItineraryFares.java

```diff
@@ -1,42 +1,64 @@
 package org.opentripplanner.routing.core;
 
-import com.google.common.collect.ArrayListMultimap;
 import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.LinkedHashMultimap;
 import com.google.common.collect.Multimap;
 import java.util.Collection;
 import java.util.HashMap;
-import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
-import org.opentripplanner.ext.fares.model.FareProduct;
+import javax.annotation.Nullable;
 import org.opentripplanner.ext.fares.model.LegProducts;
 import org.opentripplanner.framework.tostring.ToStringBuilder;
+import org.opentripplanner.model.fare.FareProduct;
+import org.opentripplanner.model.fare.FareProductUse;
 import org.opentripplanner.model.plan.Leg;
 import org.opentripplanner.transit.model.basic.Money;
 
 /**
- * <p>
- * ItineraryFares is a set of fares for different classes of users.
- * </p>
+ * ItineraryFares is a set of fares for different legs, rider categories or fare media.
  */
 public class ItineraryFares {
 
   /**
-   * A mapping from {@link FareType} to a list of {@link FareComponent}.
+   * The fare products that are valid for all legs of an itinerary, like a day pass.
+   * <p>
+   * Note: LinkedHashSet keeps the insertion order
    */
-  private final HashMap<FareType, List<FareComponent>> details = new HashMap<>();
-  private final Set<FareProduct> itineraryProducts = new HashSet<>();
-  private final Multimap<Leg, FareProduct> legProducts = ArrayListMultimap.create();
+  private final Set<FareProduct> itineraryProducts = new LinkedHashSet<>();
 
   /**
-   * A mapping from {@link FareType} to {@link Money}.
+   * Fare products that apply to one or more legs but _not_ the entire journey.
+   * <p>
+   * Note: LinkedHashMultimap keeps the insertion order
    */
-  private HashMap<FareType, Money> fare = new HashMap<>();
+  private final Multimap<Leg, FareProductUse> legProducts = LinkedHashMultimap.create();
+
+  /**
+   * The fares V1 fare "components" that apply to individual legs (not the entire price of the
+   * itinerary).
+   * <p>
+   * This is an ill-thought-out concept that was bolted onto the existing implementation in 2016 and
+   * is going to be removed once HSL has migrated off it.
+   * <p>
+   * Note: LinkedHashMultimap keeps the insertion order
+   */
+  @Deprecated
+  private final Multimap<FareType, FareComponent> components = LinkedHashMultimap.create();
+
+  /**
+   * Holds the "fares" for the entire itinerary. The definition of a fare is not clear so
+   * this is deprecated.
+   */
```

Comment:
Add JavaDoc: `@deprecated - Why?`

Reply:
Documented in 249df1651735ad50995298d5c4626be27586977c